src/client/texturesource.cpp

```diff
@@ -2391,82 +573,21 @@ video::ITexture* TextureSource::getNormalTexture(const std::string &name)
 	return NULL;
 }
 
-namespace {
-	// For more colourspace transformations, see for example
-	// https://github.com/tobspr/GLSL-Color-Spaces/blob/master/ColorSpaces.inc.glsl
-
-	inline float linear_to_srgb_component(float v)
-	{
-		if (v > 0.0031308f)
-			return 1.055f * powf(v, 1.0f / 2.4f) - 0.055f;
-		return 12.92f * v;
-	}
-	inline float srgb_to_linear_component(float v)
-	{
-		if (v > 0.04045f)
-			return powf((v + 0.055f) / 1.055f, 2.4f);
-		return v / 12.92f;
-	}
-
-	v3f srgb_to_linear(const video::SColor col_srgb)
-	{
-		v3f col(col_srgb.getRed(), col_srgb.getGreen(), col_srgb.getBlue());
-		col /= 255.0f;
-		col.X = srgb_to_linear_component(col.X);
-		col.Y = srgb_to_linear_component(col.Y);
-		col.Z = srgb_to_linear_component(col.Z);
-		return col;
-	}
-
-	video::SColor linear_to_srgb(const v3f col_linear)
-	{
-		v3f col;
-		col.X = linear_to_srgb_component(col_linear.X);
-		col.Y = linear_to_srgb_component(col_linear.Y);
-		col.Z = linear_to_srgb_component(col_linear.Z);
-		col *= 255.0f;
-		col.X = core::clamp<float>(col.X, 0.0f, 255.0f);
-		col.Y = core::clamp<float>(col.Y, 0.0f, 255.0f);
-		col.Z = core::clamp<float>(col.Z, 0.0f, 255.0f);
-		return video::SColor(0xff, myround(col.X), myround(col.Y),
-			myround(col.Z));
-	}
-}
-
 video::SColor TextureSource::getTextureAverageColor(const std::string &name)
 {
 	video::IVideoDriver *driver = RenderingEngine::get_video_driver();
-	video::SColor c(0, 0, 0, 0);
 	video::ITexture *texture = getTexture(name);
 	if (!texture)
-		return c;
+		return {0, 0, 0, 0};
 	video::IImage *image = driver->createImage(texture,
```

Comment:
btw: this function is also totally the wrong approach.
It is downloading the texture from the GPU to then determine the average color when instead we could simply calculate and remember the color before the images are uploaded to the GPU (turned into `ITexture`).

Reply:
Since currently, only source images get cached and no composed images, there is no easy fix.
So I will leave it as is for this PR.

`getTextureAverageColor` is only used for `minimap_color` anyway.